### PR TITLE
Title and description for wish lists

### PR DESCRIPTION
**Pull request by 48klocs**
I added support for title and description to wish lists, since we may want to add UI support for selecting pre-made lists from a canonical location as well as from files at some point.

here now the wish list metadata

### CODE REVIEW

File: src/app/curated-rolls/curatedRoll.ts [hunk 1]
```diff
@@ -28,3 +28,9 @@ export interface CuratedRoll {
    */
   isExpertMode: boolean;
 }
+
+export interface CuratedRollsAndInfo {
+  curatedRolls: CuratedRoll[];
+  title: string | undefined;
+  description: string | undefined;
```
Comment (bhollis) [inline src/app/curated-rolls/curatedRoll.ts]: You can write this as:

```
description?: string;
```

File: src/app/shell/rating-mode/RatingMode.tsx [hunk 2]
```diff
@@ -51,6 +52,22 @@ function mapStateToProps(state: RootState): StoreProps {
   };
 }
 
+function getTitleAndDescriptionDisplay(curatedRollsAndInfo: CuratedRollsAndInfo): string {
```
Comment (bhollis) [inline src/app/shell/rating-mode/RatingMode.tsx]: `_.compact([curatedRollsAndInfo.title, curatedRollsAndInfo.description]).join('\n')`

File: src/app/curated-rolls/curated-roll-metadata-reader.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * and the first line that looks like a description will be returned.
+ */
+export function getDescription(sourceFileText: string): string | undefined {
+  if (!sourceFileText) {
+    return undefined;
+  }
+
+  const sourceFileLineArray = sourceFileText.split('\n').slice(0, 20);
+
+  return sourceFileLineArray.map(findDescription).find((s) => s);
```
Comment (bhollis) [inline src/app/curated-rolls/curated-roll-metadata-reader.ts]: `.find(Boolean)`

**Comment by 48klocs** [comment]
Ah, good calls (didn't know about that `find` shortcut) - I'll try to get them in soon. Thanks!

**Comment by 48klocs** [comment]
Code review notes are in, and a little reducer name change to be a little more consistent (I can further audit out calling it curations in favor of wish lists in this branch, but I kind of figured I'd do that in a future branch after this lands).

I wasn't exactly sure how to handle transitioning from one shape of data (an array) to another (`CuratedRollsAndInfo`) in the load function.

**Pull request merged**